contracts/src/OvalUnlockedFeedLib.sol

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.25;
+
+import {IChainlinkSourceAdapter} from "./IChainlinkSourceAdapter.sol";
+import {Math} from "openzeppelin-contracts/contracts/utils/math/Math.sol";
+import {
+    AggregatorV3Interface,
+    ChainlinkDataFeedLib
+} from "morpho-blue-oracles/morpho-chainlink/libraries/ChainlinkDataFeedLib.sol";
+
+/// @title OvalUnlockedFeedLib
+/// @notice Library exposing unlocked price from Oval Oracle
+library OvalUnlockedFeedLib {
```

Comment:
This contract is really intresting and highlights some complexities when actually integrating as a searcher with Oval: if you queried against the oval RPC you'd be able to see the unlocked value before the lock window has passed but as the window blocks canonical reads we need to have a special way of viewing the unlocked price.

point being, in the future we should consider ways of simplifying this flow for liquidators. yes, you can simply add the RPC to your bot but the reality is actually fetching liquidation information is a bit more complex as you effectively get different responses against the contracts depending on who you ask (Oval node vs infura return diffrent prices due to oval node unlock).

one alternative (not saying we should do this) would be for the liquidator to simulate the unlock transaction form the unlocked before trying to fetch the price. this would enable them to do this without the contract change. effectively, before fetching any price first try append the unlock tx then fetch liquidation information.

Reply:
I like the approach of doing `stateDiff` override on `eth_call` when fetching prices. Though it requires some additional offchain logic to match morpho oracle to all their base1/2/quote/1/2 feed oval instances that need to have unlock override. But I recognize we might need to implement it anyways as we require adding them to Oval headers to indicate multiple unlocks are required.